Added ipynb tutorial

ojeda-e opened the pull request:
This PR fixes #63 

- [x] Added tutorial for membrane-only systems.

I would appreciate some feedback for this tutorial while I finish the other two notebooks. @lilyminium @orbeckst @richardjgowers @IAlibay 

Thanks!   
Edit: I forgot to mention I couldn't manage to make the NGLwidget render, so I ended up adding the png files of the widgets. Hope that works as an alternative.

lilyminium commented on docs/conf.py:
```diff
@@ -51,7 +51,9 @@
-    'IPython.sphinxext.ipython_directive'
+    'IPython.sphinxext.ipython_directive',
+    'sphinx_rtd_theme',
+    'nbsphinx'
```
You can add the below code to require JS for your notebooks:

```
nbsphinx_prolog = r"""
.. raw:: html

    <script src='http://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js'></script>
    <script>require=requirejs;</script>

"""
```

lilyminium replied:
Please add `nbsphinx` to the `docs/requirements.yaml` so RTD can build :)

lilyminium reviewed (changes requested):
Hmm. I don't see the review comments I left on the notebook, either here or ReviewNB, so... hope they got through. I was mostly asking for clarification on things like why you chose `wrap=True` and the number of bins -- imagine a new PhD student has been emailed your notebook from someone else, and has not been following along this project or been doing membrane research a long time :-)

Regarding the NGLView visualization. Do you mean it doesn't even render for you in your browser, or that it doesn't render on ReviewNB? I don't believe ReviewNB will ever render it, as it doesn't support JavaScript; however, if you can get ReadTheDocs to build, you could view the widgets there.

**Rendering in browser**

My foolproof way to success has been:

* install nglview into my Python environment (`conda install -y -c conda-forge nglview  && jupyter-nbextension enable nglview --py --sys-prefix`)
* Make sure I have a kernel for that specific environment (`conda activate $kernelname && python -m ipykernel install --user --name $kernelname --display-name "Python (${kernelname})"`)
* *in the Python environment that I have installed nglview into*, launch Jupyter <-- this seems to be the key step that people who can't render nglview do not do
* Make sure the notebook I make uses the new kernel with nglview I have just created

After you have rendered the NGL widgets and see it in your notebook, you then need to click Widgets > Save Notebook Widget State  for the widget to get saved into the notebook. That way Sphinx and ReadTheDocs will be able to make the interactive viewer for you.

ojeda-e commented:
> Hmm. I don't see the review comments I left on the notebook, either here or ReviewNB, so... hope they got through. I was mostly asking for clarification on things like why you chose `wrap=True` and the number of bins -- imagine a new PhD student has been emailed your notebook from someone else, and has not been following along this project or been doing membrane research a long time :-)

Oh no. I didn't get any of the reviews. As you said, neither on ReviewNB nor here on git. Arf.  Thanks for those points. I'll add them with the `nbextension` added to the requirements. I completely missed adding that file in the push, silly me.

> Regarding the NGLView visualization. Do you mean it doesn't even render for you in your browser, or that it doesn't render on ReviewNB? I don't believe ReviewNB will ever render it, as it doesn't support JavaScript; however, if you can get ReadTheDocs to build, you could view the widgets there.

I mean, I can see it on my notebook, but not on the rendered docs. Maybe this is what I missed:
> After you have rendered the NGL widgets and see it in your notebook, you then need to click Widgets > Save Notebook Widget State for the widget to get saved into the notebook. That way Sphinx and ReadTheDocs will be able to make the interactive viewer for you.

Thanks @lilyminium !

lilyminium commented on docs/source/pages/Curvature_membrane-only_systems.ipynb:
```diff
@@ -0,0 +1,6574 @@
+{
```
<p>Could you please make the title more detailed? Imagine you've just been emailed this notebook by someone else with no context.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MDAnalysis/membrane-curvature/pull/64/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

pep8speaks commented:
Hello @ojeda-e! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`docs/conf.py`](https://github.com/MDAnalysis/membrane-curvature/blob/76a8907fcc74efad2952924676505d14129a816d/docs/conf.py):

> [Line 191:4](https://github.com/MDAnalysis/membrane-curvature/blob/76a8907fcc74efad2952924676505d14129a816d/docs/conf.py#L191): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

orbeckst commented:
I added comments at https://app.reviewnb.com/MDAnalysis/membrane-curvature/blob/issue63/docs/source/pages/Curvature_membrane-only_systems.ipynb/

orbeckst commented:
@lilyminium please write a blog post with the "embedding nglview howto" https://github.com/MDAnalysis/membrane-curvature/pull/64#pullrequestreview-726642879 so that I can search it when I need it...

orbeckst commented on docs/requirements.yaml:
```diff
@@ -11,5 +11,5 @@ dependencies:
-  #- pip:
-
+  - pip:
+  - nbsphinx
```
Shouldn't this bee
```suggestion
     - nbsphinx 
```
i.e., indented so that it forms a list as the dict item `{'pip': ['nbsphinx']}` ?

Don't just take this suggestion but have a look at what the exact format of the requirements file ought to be.

orbeckst reviewed (changes requested):
Please see the comments on https://app.reviewnb.com/MDAnalysis/membrane-curvature/blob/issue63/docs%2Fsource%2Fpages%2FCurvature_membrane-only_systems.ipynb%2F/file ; primarily in a tutorial I'd like to not only see _how_ to do something but also _what the results mean_ and _why_ I should make certain choices. 

Imagine you're mentoring three undergraduate or graduate student and you send them to your package. What would you want them to learn? (And you don't want to explain everything to all 3 students 3 times so you _really_ hope the package author (past you...) has done a good job conveying everything necessary.) 

It's not only ok but very much encouraged to cite academic papers for in-depth follow up or really any reason you can think off. Citations are what makes any scholarly work possible and I see your tutorial as scholarly work.

lilyminium commented:
Having looked into that rotation state not persisting when the notebook is turned into HTML, I think it's just not being saved in the metadata. I've asked Hai about it here: https://github.com/nglviewer/nglview/issues/948#issuecomment-898121063

It's also not showing up in the ReadTheDocs check, though. You might need to either save your widget states again, or maybe add `nglview` into the `docs/requirements.yaml`?

lilyminium commented:
Hai suggested using your mouse to drag it a little (or drag it out and back again). This triggers the widget into updating the rotation state. When I did that on the notebook, saved the widget state and ran `make html`, the rotated view of the box rendered properly.

ojeda-e commented:
Thanks @orbeckst  and @lilyminium for all your suggestions, comments and help with the widgets. It works locally as suggested in this [comment](https://github.com/MDAnalysis/membrane-curvature/pull/64#issuecomment-898733734), so thanks @lilyminium for keeping an eye on this!

I think the changes in the new version of the tutorial are substantial. Some of the changes include:
- 5 sections.
- Brand new Universe.
- plots over frame.
- Interpretation of the plots.

I just want to add a couple of comments here:

- I ran a simulation with a bigger membrane patch and added a short traj to the test files to use in the tutorial.

- From the previous review, it was suggested to discuss the meaning of the plots. I added the interpretation for the plots of surface, mean and gaussian for the first approach in visualization. However, when I offered a second option to plot using `imshow`, I am not writing the interpretation again. If you consider it's better to add it, although repetitive, it can be fixed. :) 

- Although I manage to save the notebook with the widgets as suggested [here](https://github.com/MDAnalysis/membrane-curvature/pull/64#issuecomment-898733734), I'm still failing to get the widget to work properly. As you can see in the build docs, there is no widget yet. I am not sure why I get this behavior. Another strange difference is that the bullet points added in the notebook do not render for me locally, but they show up in the build. I get `WARNING: nbsphinx_widgets_path not given and ipywidgets module unavailable
looking for now-outdated files... none found` in the [build](https://readthedocs.org/projects/membrane-curvature/builds/14499499/). I tried adding but it didn't work, the warning doesn't go away.
```
nbsphinx_requirejs_path = ""
nbsphinx_widgets_path = ""
```
Thanks

lilyminium commented:
@ojeda-e does adding ipywidgets and nglview to `requirements.yaml` solve the issue?

ojeda-e commented:
> @ojeda-e does adding ipywidgets and nglview to `requirements.yaml` solve the issue?

Yes!, thanks!

ojeda-e commented:
Thanks @lilyminium for your help to make my widgets work in the tutorial! 
The first tutorial was updated and it's ready for review. Maybe @orbeckst can take a look as well :) 

Thanks

lilyminium commented:
This is looking very nice! I very much like all the pictures and visualizations you've put in. You've put a lot of effort into making a beautiful tutorial that is also comprehensive for people who are new both to the specific package, and to the analysis concept itself.

I didn't have much time to go into detail, but I did find a couple of typos. I think I remember that you did not want to put tick labels on your color bars, but I do feel that it means the plots lack some context as to the degree of curvature shown. 

I'll just approve for now to not be blocking.
